### DISCUSSION TITLE
Add proba.MultivariateGaussian

### DISCUSSION THREAD

**Discussion by MarekWadinger** [online-ml/river#1297] [Ideas]
Hello everyone! 👋

I'm currently working on an exciting project called [Unsupervised Online Outlier Detection Service with Dynamic Process Limits](https://github.com/MarekWadinger/online_outlier_detection). During my development journey, I came across the fantastic rivers beautiful API, which has been instrumental in my progress.

As the project evolved, I recognized the need to introduce support for multivariate Gaussian distribution. After exploring rivers' API, I found it to be straightforward and aligned with my requirements. To address this, I implemented [proba.MultivariateGaussian](https://github.com/MarekWadinger/river/blob/6fe74defdd6d38021da64f3be8087f5f3751eb85/river/proba/gaussian.py#L100) function.

Now, I believe this addition could be beneficial to the broader community, and I'd like to discuss whether it aligns with the library's broader goals and objectives.

You can find my fork of river with the new feature [here](https://github.com/MarekWadinger/river). I have taken care to ensure it passes the pre-commit tests and is thoroughly documented.

I welcome your valuable input and feedback on this implementation. If it proves to be valuable, I'd be thrilled to make it a part of the official library to assist others in their projects.

Thank you all for your time and consideration. Looking forward to our discussion!

**Comment by MaxHalford** [comment]
Hey hey 👋

> I'm currently working on an exciting project called [Unsupervised Online Outlier Detection Service with Dynamic Process Limits](https://github.com/MarekWadinger/online_outlier_detection).

I like the simplicity of your method! I was expecting to see a citation of the River paper though 🙈

> During my development journey, I came across the fantastic rivers beautiful API, which has been instrumental in my progress.

Kind words.

> As the project evolved, I recognized the need to introduce support for multivariate Gaussian distribution. After exploring rivers' API, I found it to be straightforward and aligned with my requirements. To address this, I implemented [proba.MultivariateGaussian](https://github.com/MarekWadinger/river/blob/6fe74defdd6d38021da64f3be8087f5f3751eb85/river/proba/gaussian.py#L100) function.

I think it's something valuable to have in River. It would be the first multivariate (continuous) distribution, so I would be inclined to introduce a new base class. There are also a couple of details which could be improved in terms of design. In any case, I encourage you to open a pull request. I can provide some comments.

**Reply by MarekWadinger** [reply]
Hey Max,

I apologize for the oversight in not including a citation to the River paper. Being already published, I have now added a reference to River at the top of the README of my project, at least. I will make sure to add the proper citation of River in future papers.

I'm opening a pull request with the addition of the MultivariateGaussian class. Will look forward to your comments to further improve the design. 😌

**Reply by MaxHalford** [reply]
> I apologize for the oversight in not including a citation to the River paper. Being already published, I have now added a reference to River at the top of the README of my project, at least. I will make sure to add the proper citation of River in future papers.

Don't apologize, I was just teasing :)
